pkg/tasks/worker/worker_pool.go

```diff
@@ -0,0 +1,116 @@
+package worker
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	m "github.com/content-services/content-sources-backend/pkg/instrumentation"
+	"github.com/content-services/content-sources-backend/pkg/tasks/queue"
+	"github.com/rs/zerolog"
+	"github.com/rs/zerolog/log"
+)
+
+type TaskHandler func(ctx context.Context, task *queue.TaskInfo) error
+
+type TaskWorkerPool interface {
+	// StartWorkers Starts workers up to number numWorkers defined in config.
+	// Should be run as a go routine.
+	StartWorkers()
```

Comment:
`TaskWorkerPool` is the interface used to interface with the workers. It represents a group of workers, where the number of workers is configurable, and each worker processes one task at a time.

I separated `worker` from `WorkerPool` to potentially allow more control over the individual worker. I had an idea to restart the worker automatically if it panicked, but I could not get that to work as expected. My other concern with automatically restarting workers is that they'll just continue to panic. Instead, I tried to write the worker loop such that panics are caught and the worker doesn't exit.